nitro_rpc/src/NitroRPC.sol

```diff
@@ -1,23 +1,82 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.17;
 
+
+import {INitroTypes} from "nitro/interfaces/INitroTypes.sol";
+import {NitroUtils} from "nitro/libraries/NitroUtils.sol";
+import {IForceMoveApp} from "nitro/interfaces/IForceMoveApp.sol";
+
 /**
  * @title NitroRPC
  * @dev Solidity implementation of Nitro RPC protocol structures
  */
-interface NitroRPC {
-
+contract NitroRPC is IForceMoveApp {
     struct Payload {
         uint256 requestId;
+        uint256 timestamp;
         string method;
         bytes params;
         bytes result;
-        uint256 timestamp;
     }
 
     struct PayloadSigned {
       Payload rpcMessage;
-      INitroTypes.Signature clientSig
-      INitroTypes.Signature serverSig
+      INitroTypes.Signature clientSig;
+      INitroTypes.Signature serverSig;
+    }
+
+    enum AllocationIndices {
+        Client,
+        Server
+    }
+
+    /**
+     * @notice Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @dev Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @param fixedPart Fixed part of the state channel.
+     * @param proof Array of recovered variable parts which constitutes a support proof for the candidate.
+     * @param candidate Recovered variable part the proof was supplied for.
+     */
+    function stateIsSupported(
+        INitroTypes.FixedPart calldata fixedPart,
+        INitroTypes.RecoveredVariablePart[] calldata proof,
+        INitroTypes.RecoveredVariablePart calldata candidate
+    ) external pure override returns (bool, string memory) {
+        require(fixedPart.participants.length == uint256(AllocationIndices.Server) + 1, "bad number of participants");
+        
+        PayloadSigned memory payloadSigned = abi.decode(candidate.variablePart.appData, (PayloadSigned));
+        requireValidPayload(fixedPart, payloadSigned);
+
+        return (true, "");
```

Comment:
Do we have any other checks pertaining to `Payload`? Should payloads be chained, in which case we should introduce `proof` checks?